build/scripts-3.11/import.py

```diff
@@ -0,0 +1,191 @@
+    parser.add_argument('-v', '--verbose', action='store_true')
+    parser.add_argument('expr')
+    args = parser.parse_args()
+
+    level = logging.INFO
+    if args.verbose:
+        level = logging.DEBUG
+    logging.basicConfig(level=level)
+
+    with open(os.path.expanduser(args.config), 'r') as fo:
+        config = yaml.load(fo, Loader=yaml.SafeLoader)
+
+    # db = Database(config, args.dbname).connect()
+
+    # iterate over matching files on the filesystem
+    for f in glob.iglob(args.expr, recursive=True):
+        print(f)
+    #     dirname = os.path.dirname(f)
+    #     basename = os.path.basename(f)
+    #     # probe for dpdash-compatibility and gather information
+    #     probe = dpimport.probe(f)
+    #     if not probe:
+    #         logger.debug('document is unknown %s', basename)
+    #         continue
+    #     # nothing to be done
+    #     if db.exists(probe):
+    #         logger.info('document exists and is up to date %s', probe['path'])
+    #         continue
+    #     logger.info('document does not exist or is out of date %s', probe['path'])
+    #     # import the file
+    #     logger.info('importing file %s', f)
+    #     dppylib.import_file(db.db, probe)
+
+    # logger.info('cleaning metadata')
+    # lastday = get_lastday(db.db)
+    # if lastday:
+    #     clean_metadata(db.db, lastday)
+
+def clean_metadata(db, max_days):
+    studies = col.defaultdict()
+    subjects = list()
+
+    for subject in max_days:
+        if subject['_id']['study'] not in studies:
+            studies[subject['_id']['study']] = {}
+            studies[subject['_id']['study']]['subject'] = []
+            studies[subject['_id']['study']]['max_day'] = 0
+
+            # if there are more than 2, drop unsynced
+            metadata = list(db.metadata.find(
+                {
+                    'study' : subject['_id']['study']
+                },
+                {
+                    '_id' : True,
+                    'collection' : True,
+                    'synced' : True
+                }
+            ))
+
+            if len(metadata) > 1:
+                for doc in metadata:
+                    if doc['synced'] is False and 'collection' in doc:
+                        db[doc['collection']].drop()
+                    if doc['synced'] is False:
+                        db.metadata.delete_many(
+                            {
+                                '_id': doc['_id']
+                            }
+                        )
+
+        subject_metadata = col.defaultdict()
+        subject_metadata['subject'] = subject['_id']['subject']
+        subject_metadata['synced'] = subject['synced']
+        subject_metadata['days'] = subject['days']
+        subject_metadata['study'] = subject['_id']['study']
+
+        studies[subject['_id']['study']]['max_day'] = studies[subject['_id']['study']]['max_day'] if (studies[subject['_id']['study']]['max_day'] >= subject['days'] ) else subject['days']
```

Comment:
This feels like too much for one line

Reply:
this is outdated it's been removed
[655c34d](https://github.com/AMP-SCZ/dpimport/pull/17/commits/655c34d7cc922806cb0928f4d5234932a18bcfaf)